src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -1232,6 +1251,22 @@ bool ITokenCacheInternal.HasTokensNoLocks()
             return Accessor.HasAccessOrRefreshTokens();
         }
 
+        internal /* internal for test only */ void RemoveOboAccountInternal(string partitionKey, RequestContext requestContext)
+        {
+            ILoggerAdapter logger = requestContext.Logger;
+
+            //Filter and remove tokens based on OBO Cache Key
+            var refreshTokens = Accessor.GetAllRefreshTokens(partitionKey);
+            refreshTokens.RemoveAll(item => !item.OboCacheKey.Equals(partitionKey, StringComparison.OrdinalIgnoreCase));
+
+            RemoveRefreshTokens(refreshTokens, logger, out bool filterByClientId);
+
+            var accessTokens = Accessor.GetAllAccessTokens(partitionKey);
+            accessTokens.RemoveAll(item => !item.OboCacheKey.Equals(partitionKey, StringComparison.OrdinalIgnoreCase));
+
+            RemoveAccessTokens(accessTokens, logger, filterByClientId);
+        }
```

Comment:
I think you can now check if there are any access tokens / refresh tokens left in the node. If there aren't, I think it's ok to delete everything else.

Reply:
Discussed offline: It is fine to leave the tokens there, the user developer expirence is not affected much and the tokenless accounts should be handled by cache eviction which is strongly encouraged by our docs.